pkg/stub/handler.go

```diff
@@ -863,27 +962,29 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		if dockercfgSecret.Name != v1alpha1.SamplesRegistryCredentials {
 			return nil
 		}
-		h.mutex.Lock()
-		defer h.mutex.Unlock()
 
 		srcfg, _ := h.sdkwrapper.Get(v1alpha1.SamplesResourceName)
 		if srcfg != nil {
-			doit, err := h.ProcessManagementField(srcfg)
-			if !doit || err != nil {
-				return err
+			switch srcfg.Spec.ManagementState {
+			case operatorsv1alpha1api.Removed:
+				logrus.Debugln("Ignoring secret event because operator is in removed state")
```

Comment:
don't you still need to process the secret deleted event if the samples resource is in the removed state?  the earlier comments implied that was how the credentialsexist status would get updated.  (though you also need to ensure on a resync we can detect the credentials are missing and update the status, in case you miss the delete event)

(also it's the samples resource, not the operator, that is removed/unmanaged).

Reply:
I took Removed as still being !Managed, or Unmanaged, but with additional effects.  Also, the removed processing removes the secret.  So if the user, after marking Removed, first adds the secret, and then deletes it, we would not have created the secret in the openshift namespace in the first place.  

Not sure what you exactly mean by "on a resync..."

Fixed debug string to say samples resource.

Reply:
@gabemontero https://github.com/openshift/cluster-samples-operator/pull/54/files#diff-2d7ceed365f1ea47bba386dac808cb07R906

I interpret that comment as meaning the flow will be:

0) CredentialsExist=true, secret exists.
1) user sets spec.state=Removed
2) cleanupopenshiftnamespace gets run (this deletes the secret)
4) we will see the delete event for the secret
5) we will ignore it because src.spec.ManagementState is removed, so the CredentialsExist status never gets updated to false.

As for what I mean about on a resync, i mean suppose we miss the even in (4) (and assume you fix the logic in (5)).  If we miss the delete event, how does CredentialsExist ever get updated to false?

Reply:
OK ... @bparees and I talked 
- will fix comment
- will set cred exists condition to false when we delete since we are deleting both namespace secrets
- add a TODO for deciding what polling or other alternative is needed on missing secret delete event since we cannot add finalizer

Reply:
we should mark the one in the operator namespace as to whether we created it or not (we will have created it if we created it by copying pull secrets from the installer).

and then we should delete it if we own it, otherwise leave it.